QMigrate/lut/lut.py

```diff
@@ -369,6 +373,21 @@ class LUT(Grid3D):
 
     """
 
+    def __init__(self, fraction_tt=0.1, lut_file=None, **grid_kwargs):
+        """Instantiate the LUT object."""
+
+        if grid_kwargs:
+            super().__init__(**grid_kwargs)
+            self.fraction_tt = fraction_tt
+            self.maps = {}
+            self.phases = []
+            self.velocity_model = ""
+        else:
```

Comment:
Flag what? Nothing there is deprecated, it's a new feature.

Reply:
I mean that the `else:` is handling the old (deprecated) method of reading in a LUT.

Reply:
Ah, not quite - it's a (jank) way of allowing the function `read_lut` to instantiate a completely empty LUT (rather than having to mock the grid definition, then update). The deprecated part is the:

```python
self.phase = ["P", "S"]
if lut_file is not None:
    self.load(lut_file)
```

for which there is a message to the user in that method.

My suggested change:

```python
    def __init__(self, fraction_tt=0.1, lut_file=None, **grid_kwargs):
        """Instantiate the LUT object."""

        if grid_kwargs:
            super().__init__(**grid_kwargs)
            self.maps = {}
            self.phases = []
            self.velocity_model = ""
        elif lut_file is not None:
            self.phases = ["P", "S"]  # Handle old lookup tables
            self.load(lut_file)
        self.fraction_tt = fraction_tt
```

I need to just have a think to make sure this will handle all of the cases (back to the Venn diagram...)